activesupport/test/message_verifier_test.rb

```diff
@@ -36,13 +38,25 @@ def test_simple_round_tripping
   end
 
   def test_verified_returns_false_on_invalid_message
-    assert !@verifier.verified("purejunk")
+    message = @verifier.generate(@data)
+    mock = MiniTest::Mock.new
+    mock.expect(:call, false, [message])
+    @verifier.stub(:valid_message?, mock) do
+      assert !@verifier.verified(message)
+    end
+    mock.verify
```

Comment:
I don't think we need to duplicate the cases here. @matthewd what do you think about this?